libraries/chef_ingredient_resource.rb

```diff
@@ -45,6 +45,11 @@ class ChefIngredient < Chef::Resource::LWRPBase
       # Attribute to enable selecting packages built for earlier versions in
       # platforms that are not yet officially added to Chef support matrix
       attribute :platform_version_compatibility_mode, kind_of: [TrueClass, FalseClass], default: false
+
+      # Atttibutes for configuring a specific platform package
+      attribute :platform, kind_of: String
+      attribute :platform_version, kind_of: String
+      attribute :architecture, kind_of: String
```

Comment:
Let's try making this:

```ruby
attribute :architecture, kind_of: String, equal_to: Mixlib::Install::Options::SUPPORTED_ARCHITECTURES
```

Likewise we can improve a few more attributes validation logic:

```ruby
attribute :product_name, kind_of: String, name_attribute: true, equal_to: Mixlib::Install::Options:: SUPPORTED_PRODUCT_NAMES
attribute :channel, kind_of: Symbol, default: :stable, equal_to: Mixlib::Install::Options::SUPPORTED_CHANNELS
```

We'll probably need something like this at the top of the file:

```ruby
begin
  require 'mixlib/install'
rescue LoadError
  ensure_mixlib_install_gem_installed!
end
```